wire-ios-sync-engine/Source/SessionManager/SessionManager.swift

```diff
@@ -519,14 +519,27 @@ public final class SessionManager: NSObject, SessionManagerType {
             self.notificationsTracker = nil
         }
 
-        let analyticsConfig = analyticsSessionConfiguration.map {
+        let analyticsConfig = analyticsServiceConfiguration.map {
             AnalyticsService.Config(
-                secretKey: $0.countlyKey,
-                serverHost: $0.host
+                secretKey: $0.secretKey,
+                serverHost: $0.serverHost
             )
         }
 
-        analyticsService = AnalyticsService(config: analyticsConfig)
+        analyticsService = AnalyticsService(
+            config: analyticsConfig,
+            logger: { WireLogger.analytics.debug($0) }
+        )
+
+        if analyticsServiceConfiguration?.didUserGiveTrackingConsent == true {
+            Task { [analyticsService] in
+                do {
+                    try await analyticsService.enableTracking()
+                } catch {
+                    WireLogger.analytics.error("failed to enable tracking: \(error)")
+                }
+            }
+        }
```

Comment:
I guess that's the earliest place or only place where we can start this Task at the moment